dapr/proto/scheduler/v1/scheduler.proto

```diff
@@ -0,0 +1,158 @@
+syntax = "proto3";
+
+package dapr.proto.scheduler.v1;
+
+import "google/protobuf/any.proto";
+
+option go_package = "github.com/dapr/dapr/pkg/proto/scheduler/v1;scheduler";
+
+service Scheduler {
+	// ScheduleJob is used by the daprd sidecar to schedule a job.
+	rpc ScheduleJob(ScheduleJobRequest) returns (ScheduleJobResponse) {}
+	// Get a job
+	rpc GetJob(GetJobRequest) returns (GetJobResponse) {}
+	// DeleteJob is used by the daprd sidecar to delete a job.
+	rpc DeleteJob(DeleteJobRequest) returns (DeleteJobResponse) {}
+	// WatchJobs is used by the daprd sidecar to connect to the Scheduler
+	// service to watch for jobs triggering back.
+	rpc WatchJobs(stream WatchJobsRequest) returns (stream WatchJobsResponse) {}
+}
+
+message Job {
+  // The schedule for the job.
+  optional string schedule = 1;
+
+  // Optional: jobs with fixed repeat counts (accounting for Actor Reminders).
+  optional uint32 repeats = 2;
+
+  // Optional: sets time at which or time interval before the callback is invoked for the first time.
+  optional string due_time = 3;
+
+  // Optional: Time To Live to allow for auto deletes (accounting for Actor Reminders).
+  optional string ttl = 4;
+
+  // Job data.
+  google.protobuf.Any data = 5;
+}
+
+// TypeJob is the message used by the daprd sidecar to schedule a job
+// from an App.
+message TypeJob {}
+
+// TypeActorReminder is the message used by the daprd sidecar to
+// schedule a job from an Actor Reminder.
+message TypeActorReminder {
+  // id is the actor ID.
+	string id = 1;
+
+  // type is the actor type.
+	string type = 2;
+}
+
+// JobMetadataType holds the typed metadata associated with the job for
+// different origins.
+message JobMetadataType {
+  oneof type {
+    TypeJob job = 1;
+    TypeActorReminder actor = 2;
+  }
+}
+
+// JobMetadata is the message used by the daprd sidecar to schedule/get/delete a
+// job.
+message JobMetadata {
+  // app_id is the App ID of the requester.
+	string app_id = 1;
+
+  // namespace is the namespace of the requester.
+  string namespace = 2;
+
+  // type is the type of the job.
+	JobMetadataType type = 3;
+}
+
+// WatchJobsRequest is the message used by the daprd sidecar to connect to the
+// Scheduler and send Job process results.
+message WatchJobsRequest {
+  oneof watch_job_request_type {
+    WatchJobsRequestInitial initial = 1;
+    WatchJobsRequestResult result = 2;
+  }
+}
+
+// WatchJobsRequestInitial is the initial request to start watching for jobs.
+message WatchJobsRequestInitial {
+  // app_id is the App ID of the requester.
+  string app_id = 1;
+
+  // namespace is the namespace of the requester.
+  string namespace = 2;
+
+  // actorTypes is the optional list of actor types to watch for.
+  repeated string actorTypes = 3;
+}
+
+// WatchJobsRequestResult is the result of a job execution to allow the job to
+// be marked as processed.
+message WatchJobsRequestResult {
+  // uuid is the uuid of the job that has finished processing.
+  uint64 uuid = 1;
```

Comment:
uuid is 128 bits so can't fit in uint64?

Reply:
This is used as an incremental counter, not really a `uuid`. I can change the name to be job `id` and put an explicit note in the comment description for it, or do you have a suggestion for a different name?

Reply:
Yes please, rename it. `id` or `seq` (sequence number) could work. The name `uuid` implies it's a UUID which is not a number